www/waterfall_view/src/module/main.module.coffee

```diff
@@ -23,6 +23,19 @@ class Waterfall extends Controller
             icon: "search-minus"
             action: @zoomMinus
         ]
+
+        # 'waterfall' class needs to be dynamically added to the body in order 
+        #  to support waterfall-specific styling of the body.  (this is a bit
+        #  awkward since the body is provided by guanlecoja-ui and is the same
+        #  element as you switch between different plugin pages, therefore the
+        #  class needs to removed upon exiting the waterfall via the $destroy
+        #  event below.)
+        body = @$rootElement.find("body")
+        body.addClass("waterfall")
```

Comment:
I would rather call that class hundredpercent (so that other dashboard can use the same trick.)

Reply:
Sounds good.  I'll make that change.

Reply:
I'll move it the hundredpercent style into somewhere more global as well I suppose.  `/buildbot/www/base/src/styles/styles.less` is what I was thinking.

Reply:
@tardyp, good thinking: You were right to be suspicious of the css rule name `body.waterfall` since there was already a rule for `.waterfall` below that, so my `body.waterfall` was inadvertently picking up the rules that were meant for the the `div.waterfall`, and in fact that was what was making it work.  Once I renamed it to `hundredpercent` it stopped working again -- I realized setting `body { height : 100% }` does nothing without also setting `html { height : 100% }` which was the other rule that I removed the `guanlecoja-ui`.  

However, making the new `.hundredpercent` rule use { height : 100vh } seems to work well.  When I set `100vh` on the body it does not cause the "disappearing upon opening dialog" problem that happened when setting `100vh` on the `div.waterfall`

On a side note, whenever I try to add my new changes with `git add -A .`, I end up accidentally adding a bunch of package-lock.json files that apparently aren't already in the repository.  Should we add that to .gitignore or should we just start adding those files to the repo?  Internet says you should add them to the repo, but I don't know much about it or have an opinion about it.

Reply:
we do commit the yarn.lock from time to time, but imho this is just noise.